### PR TITLE
refactor: db-level sorting

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **New Features**
  * DB-backed due-date sorting for tasks, placing items without due dates last and honoring pagination.

* **Improvements**
  * Sorting now respects filters (status, category, labels, priorities, starred/private) and reduces in-memory re-sorting for better performance and reliability.

* **Tests**
  * Added/updated tests for due-date ordering, filtering, pagination, and edge cases (empty ids, starred filtering).

* **Chores**
  * Package build number incremented.

### CODE REVIEW

File: lib/database/database.dart [hunk 1]
```diff
@@ -1105,6 +1105,172 @@ class JournalDb extends _$JournalDb {
...
+          variables.add(Variable<String>(id));
+          placeholders.add('?${variables.length}');
+        }
+        conditions.add(
+          r"json_extract(serialized, '$.data.categoryId') "
+          'IN (${placeholders.join(", ")})',
+        );
+      }
+      buf.write('AND (${conditions.join(" OR ")}) ');
+    }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/database/database.dart:1243]: Fixed. Replaced the incorrect `json_extract(serialized, '$.data.categoryId')` with proper `labeled` table subqueries: `NOT EXISTS (SELECT 1 FROM labeled WHERE journal_id = journal.id)` for unlabeled, and `EXISTS (SELECT 1 FROM labeled WHERE journal_id = journal.id AND label_id IN (...))` for specific labels. Also added `labeled` to `readsFrom`.

File: lib/database/database.dart [hunk 2]
```diff
@@ -1105,6 +1105,172 @@ class JournalDb extends _$JournalDb {
...
+    required List<String> categoryIds,
+    List<String>? labelIds,
+    List<String>? priorities,
+    List<String>? ids,
+    int limit = 500,
+    int offset = 0,
+  }) {
+    if (taskStatuses.isEmpty || categoryIds.isEmpty) {
+      return Future.value([]);
+    }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/database/database.dart]: Fixed. Added `(ids != null && ids.isEmpty)` to the early-return guard so an empty FTS hit list correctly returns no results instead of falling through to the full task list.

File: lib/database/database.dart [hunk 3]
```diff
@@ -1105,6 +1105,172 @@ class JournalDb extends _$JournalDb {
...
+
+    final variables = <Variable<Object>>[];
+    final buf = StringBuffer()
+      ..write(
+        'SELECT * FROM journal '
+        'INDEXED BY idx_journal_tasks_due_active ',
+      )
+      ..write("WHERE type = 'Task' AND deleted = 0 ")
+      // Task statuses
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/database/database.dart:1160]: Fixed. Added `AND task = 1` to the WHERE clause to match other task query paths and ensure index coverage.

File: lib/database/database.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/database/database.dart]: Fixed. Same fix as the Gemini comment above — replaced `json_extract(serialized, '$.data.categoryId')` with proper `labeled` table subqueries (`NOT EXISTS`/`EXISTS`) and added `labeled` to `readsFrom`.
